examples/shader/post_processing.rs

```diff
@@ -162,20 +78,249 @@ fn main_camera_cube_rotator_system(
-    source_image: Handle<Image>,
-}
+    pub struct PostProcessingPlugin;
+
+    impl Plugin for PostProcessingPlugin {
+        fn build(&self, app: &mut App) {
+            app.add_plugin(Material2dPlugin::<PostProcessingMaterial>::default())
+                .add_system(setup_new_color_blindness_cameras)
+                .add_system(update_image_to_window_size)
+                .add_system(update_enable);
+        }
+    }
+
+    /// To support window resizing, this fits an image to a windows size.
+    #[derive(Component)]
+    struct FitToWindowSize {
+        image: Handle<Image>,
+        window_id: WindowId,
+    }
+    #[derive(Component)]
+    pub struct PostProcessingCamera;
+
+    /// Update image size to fit window
+    fn update_image_to_window_size(
+        windows: Res<Windows>,
+        mut image_events: EventWriter<AssetEvent<Image>>,
+        mut images: ResMut<Assets<Image>>,
+        mut resize_events: EventReader<WindowResized>,
+        fit_to_window_size: Query<&FitToWindowSize>,
+    ) {
+        for resize_event in resize_events.iter() {
+            for fit_to_window in fit_to_window_size.iter() {
+                if resize_event.id == fit_to_window.window_id {
+                    let size = {
+                        let window = windows.get(fit_to_window.window_id).expect("ColorBlindnessCamera is rendering to a window, but this window could not be found");
+                        Extent3d {
+                            width: window.physical_width(),
+                            height: window.physical_height(),
+                            ..Default::default()
+                        }
+                    };
+                    let image = images.get_mut(&fit_to_window.image).expect(
+                        "FitToScreenSize is referring to an Image, but this Image could not be found",
+                    );
+                    dbg!(format!("resize to {:?}", size));
+                    image.resize(size);
+                    // Hack because of https://github.com/bevyengine/bevy/issues/5595
+                    image_events.send(AssetEvent::Modified {
+                        handle: fit_to_window.image.clone(),
+                    });
+                }
+            }
+        }
+    }
+    fn update_enable(
+        time: Res<Time>,
+        cameras: Query<(&Handle<PostProcessingMaterial>, &PostProcessingCamera)>,
+        mut materials: ResMut<Assets<PostProcessingMaterial>>,
+    ) {
+        for (handle, camera) in &cameras {
+            let mut mat = materials.get_mut(handle).unwrap();
+
+            mat.offset_r = Vec2::new(-0.01f32 * time.seconds_since_startup().sin() as f32, 0f32);
+            mat.offset_g = Vec2::new(
+                0.01f32 * time.seconds_since_startup().sin() as f32,
+                0.01f32 * time.seconds_since_startup().cos() as f32,
+            );
+            mat.offset_b = Vec2::new(0f32, -0.01f32 * time.seconds_since_startup().cos() as f32);
+        }
+    }
```

Comment:
wrong function name, but this allows the resizing to work. I think it enables to work because it forces a re extraction of the material, hence updating the image size.

This means it breaks without an update to the material each frame.

I think a fix would be to update the material when detecting a resize within `update_image_to_window_size`.